Zimbra/zm-load-testing
# HA performance: zsoap modification and profile creation for HA performance testing

Test script
1) Add assertion for the response for request like search, getMessage to verify above data before, during and after failover.
2) Created new property file for HA test to do better search and get message queries.
3) Added email validation in LMTP test. now we will login with the recipient user to check if message is delivered or not.
4) Added way to create random message so that with LMTP we insert new message each time.
5) Validated same message with recipient user.
6) Create new folder under test to separate out the HA tests.

LMTP test: executed and verified that all test are passing. with single user and 10 users.
SOAP test: executed and verified that all test are passing

Note: all data for search is based on large data which is created for HA test, due to size issue (it exceeds 50MB) I was not able to commit the tar file.
`HATest-zsoap:
   [jmeter] Executing test plan: /home/ubuntu/sakchhi/zm-load-testing/tests/HATest/zsoap/zsoap.jmx ==> /home/ubuntu/sakchhi/zm-load-testing/logs/HATest-zsoap-requests.log
   [jmeter] Writing log file to: /home/ubuntu/sakchhi/zm-load-testing/logs/HATest-zsoap-jmeter.log



   [jmeter] Creating summariser <summary>
   [jmeter] Created the tree successfully using /home/ubuntu/sakchhi/zm-load-testing/tests/HATest/zsoap/zsoap.jmx
   [jmeter] Starting the test @ Wed Mar 17 17:28:17 UTC 2021 (1616002097341)
   [jmeter] Waiting for possible Shutdown/StopTestNow/Heapdump message on port 4445
   [jmeter] summary +     13 in 00:00:13 =    1.0/s Avg:    82 Min:     3 Max:   316 Err:     0 (0.00%) Active: 1 Started: 1 Finished: 0
   [jmeter] summary +     51 in 00:00:30 =    1.7/s Avg:    53 Min:     1 Max:   277 Err:     0 (0.00%) Active: 1 Started: 1 Finished: 0
   [jmeter] summary =     64 in 00:00:42 =    1.5/s Avg:    59 Min:     1 Max:   316 Err:     0 (0.00%)
   [jmeter] summary +     54 in 00:00:32 =    1.7/s Avg:    51 Min:     1 Max:   237 Err:     0 (0.00%) Active: 1 Started: 1 Finished: 0
   [jmeter] summary =    118 in 00:01:14 =    1.6/s Avg:    55 Min:     1 Max:   316 Err:     0 (0.00%)
   [jmeter] summary +     51 in 00:00:30 =    1.7/s Avg:    50 Min:     0 Max:   282 Err:     0 (0.00%) Active: 1 Started: 1 Finished: 0
   [jmeter] summary =    169 in 00:01:44 =    1.6/s Avg:    53 Min:     0 Max:   316 Err:     0 (0.00%)
   [jmeter] summary +     13 in 00:00:07 =    1.9/s Avg:    43 Min:     1 Max:   114 Err:     0 (0.00%) Active: 0 Started: 1 Finished: 1
   [jmeter] summary =    182 in 00:01:51 =    1.6/s Avg:    53 Min:     0 Max:   316 Err:     0 (0.00%)
   [jmeter] Tidying up ...    @ Wed Mar 17 17:30:08 UTC 2021 (1616002208222)
   [jmeter] ... end of run

BUILD SUCCESSFUL
Total time: 1 minute 52 seconds`
